packages/aws-cdk-lib/aws-eks/lib/cluster.ts

```diff
@@ -1716,6 +1787,47 @@ export class Cluster extends ClusterBase {
     this.defineCoreDnsComputeType(props.coreDnsComputeType ?? CoreDnsComputeType.EC2);
 
   }
+  /**
+   * Adds an access entry to the cluster's access entries map.
+   *
+   * If an entry already exists for the given principal, it adds the provided access policies to the existing entry.
+   * If no entry exists for the given principal, it creates a new access entry with the provided access policies.
+   *
+   * @param principal - The principal (e.g., IAM user or role) for which the access entry is being added.
+   * @param policies - An array of access policies to be associated with the principal.
+   *
+   * @throws {Error} If the uniqueName generated for the new access entry is not unique.
+   *
+   * @returns {void}
+   */
+  public addToAccessEntry(id: string, principal: string, policies: IAccessPolicy[]) {
```

Comment:
can this be a `private` function if we customers are using grantaccess()?

Reply:
yes will fix it!

Reply:
DONE